### PR TITLE
Avoid generic reduce to reduce gc pressure

### PR DESCRIPTION
**Pull request by jobergum**
Merge the two tensor.reduce calls to a single custom reduce for the exact reduce function we need for this, significantly reduces memory allocations. 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: model-integration/src/main/java/ai/vespa/embedding/SpladeEmbedder.java [hunk 1]
```diff
@@ -167,4 +158,42 @@ public void deconstruct() {
...
+
+    /**
+     * Custom max reduce over the sequence dimension of the output tensor. This
+     * to reduce GC pressure from generic Tensor.reduce operation.
+     *
+     * @param tensor the model output tensor of shape d1,dim where d1 is the sequence length and dim is size
+     *               of the vocabulary
+     * @return A tensor of shape d1,1 where each value is the max of the log of the relu of the input tensor
+     */
+    private static IndexedTensor customMaxReduceOverLogOfRelu(IndexedTensor tensor) {
```
Comment (bjorncs) [inline model-integration/src/main/java/ai/vespa/embedding/SpladeEmbedder.java:170]: Consider inlining this logic with for loop in `sparsify()` - the temporary tensor is not really needed.

File: model-integration/src/test/java/ai/vespa/embedding/SpladeEmbedderTest.java [hunk 2]
```diff
@@ -43,6 +43,19 @@ public void testZeroTokens() {
         assertEquals(0, result.size());
     }
 
+    @Test
```
Comment (bjorncs) [inline model-integration/src/test/java/ai/vespa/embedding/SpladeEmbedderTest.java:46]: Either add a timeout or ignore the test by default

**Review by bjorncs — DISMISSED** [review]

File: model-integration/src/main/java/ai/vespa/embedding/SpladeEmbedder.java (hunk 1 above)
Comment (jobergum) [reply model-integration/src/main/java/ai/vespa/embedding/SpladeEmbedder.java:170]: Great feedback!

**Review by bjorncs — APPROVED** [review]
Nice! 👍

**Pull request merged**